### PR TITLE
Return resolved `nemo` object in constructor callback

### PR DESCRIPTION
**Pull request by ShayDavidson**
_Re-opend against `develop`._

Without the nemo in the callback, you have to rely on the returned object outside the scope of the callback. This breaks the node callback convention, and prevents from 'promisifying' the Nemo function.

I've also updated the tests, README and examples to reflect this change.

@grawk this include the changes you requested, outside of reverting the documentation since I already updated the examples to reflect the change. IMO, that revert is unnecessary then. Let me know if that's okay.

### CODE REVIEW

**Comment by grawk** [comment]
Hi @ShayDavidson. The "examples" I was referring to are the ones found here: https://github.com/paypal/nemo-example-app and here: http://nemo.js.org/pages/getting-started.html

But, I think it's fine to merge this and perhaps add some additional note to the README under the first example about the constructor returning an unresolved nemo object.

:+1: from me.

File: README.md [hunk 1]
```diff
@@ -314,7 +313,10 @@ the mocha `done` function as the callback:
 var nemo;
 describe('my nemo suite', function() {
   before(function(done) {
-    nemo = Nemo(config, done);
+    Nemo(config, function(err, resolvedNemo) {
```
Comment (OuranosSkia) [inline README.md]: From a consistent style point of view, you should have a space between `function` and `(` to keep the same style the rest of the file uses.

File: test/constructor.js [hunk 2]
```diff
@@ -32,11 +31,11 @@ describe('@constructor@', function () {
 
   it("should launch nemo with @noConfigPath@overrideArg@", function (done) {
     delete process.env.nemoBaseDir;
-    nemo = Nemo({
+    Nemo({
       "driver": {
         "browser": "phantomjs"
       }
-    }, function () {
+  }, function (err, nemo) {
```
Comment (OuranosSkia) [inline test/constructor.js]: All of your spacings for where you do this are de-indented one chunk from where they should be.

File: README.md (hunk 1 above)
Comment (OuranosSkia) [inline README.md]: This comment applies across test files.

**Comment by OuranosSkia** [comment]
Aside from the few style issues I noted, everything else looks good!

**Comment by ShayDavidson** [comment]
@OuranosSkia fixed.

File: test/config.js [hunk 3]
```diff
@@ -20,7 +19,7 @@ describe('@config@', function () {
           }
         }
       }
-    }, function (err) {
+  }, function (err, nemo) {
```
Comment (OuranosSkia) [inline test/config.js]: Still have some of these spacing issues scattered about.

File: test/proxy.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ describe("@proxy@ ", function () {
           "forBrowser": ['phantomjs']
         }
       }
-    }, function (err) {
+    }, function (err, nemo) {
```
Comment (gabrielcsapo) [inline test/proxy.js]: :+1:

File: test/constructor.js [hunk 5]
```diff
@@ -3,13 +3,12 @@
 
 var assert = require('assert'),
   path = require('path'),
-  nemo = {},
```
Comment (gabrielcsapo) [inline test/constructor.js]: :+1:

File: index.js [hunk 6]
```diff
@@ -120,7 +120,7 @@ function Nemo(_basedir, _configOverride, _cb) {
         return;
       }
       _.merge(nemo, _nemo);
-      cb();
+      cb(null, nemo);
```
Comment (gabrielcsapo) [inline index.js]: :+1:

File: test/config.js (hunk 3 above)
Comment (gabrielcsapo) [inline test/config.js]: @ShayDavidson would you be able to post jshint results?

**Comment by grawk** [comment]
I'll go ahead and merge this then fix the couple remaining indentation issues locally before publishing

**Pull request merged**